### PR TITLE
feat: base integration of triggers

### PR DESCRIPTION
**Pull request by nduchak**
This PR contains a basic integration of Triggers service:
- service structure
- service precache
- adjust CLI
- add configuration
- migrations
- event handling and DB models

### CODE REVIEW

File: src/services/triggers/handlers/provider.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    if (providerService.emit) providerService.emit('created', wrapEvent('ProviderRegistered', { provider, url }))
+    logger.info(`Created new Provider with address ${provider} and url  ${url}`)
+  }
+}
+
+function isValidEvent (value: string): value is keyof typeof handlers {
+  return value in handlers
+}
```
Comment (jurajpiar) [inline src/services/triggers/handlers/provider.ts]:
```suggestion
function isValidEvent (eventName: string): eventName is keyof typeof handlers {
  return eventName in handlers
}
```

**Review by jurajpiar — APPROVED** [review]
LGTM
I'm guessing you are going to add the other fields to the triggers model, right?
Also, perhaps we should agree upon the name for this service, since the dev project calls it notifications, the cache triggers (personally I'd vote for it) and UI notifier (since notifications is already a svc for marketplace notifs).
One thing I noticed when running it is that the precache doesn't process triggers when run with the `all` arg.

Couldn't test whether it works with the UI for some problems with the contract addresses I have (locally I suspect).

**Review by 0xartem — APPROVED** [review]
In terms of coding I don't have comments, looks good. But I have troubles to fit the whole picture of interactions between different modules inside of my head, so I'm not completely clear on what to recommend as improvements. Maybe next sprint I will be able to spend more time with the code to give you more valuable feedback.

File: src/services/triggers/handlers/provider.ts (hunk 1 above)
Comment (nduchak) [reply src/services/triggers/handlers/provider.ts]: Done

**Comment by nduchak** [comment]
> LGTM
> I'm guessing you are going to add the other fields to the triggers model, right?

I will add another model for `Subscription plans` when do integration with the notifier service

> One thing I noticed when running it is that the precache doesn't process triggers when run with the `all` arg.

Hmm... Works for me. Probably some configuration issue. Try to use ganache config (NODE_ENV=ganache)

> Couldn't test whether it works with the UI for some problems with the contract addresses I have (locally I suspect).

This version probably will not work with UI as it doesn't have the required data for the provider table

**Comment by jurajpiar** [comment]
> I will add another model for Subscription plans when do integration with the notifier service

Sure, though it would have been easier for me if it did have the plans already. But, I understand if the reason for delaying this was that you don't know exactly the structure you'll be getting from the notifier service.

> Hmm... Works for me. Probably some configuration issue. Try to use ganache config (NODE_ENV=ganache)

Probably doing something else wrong then. I do use NODE_ENV=ganache.

> This version probably will not work with UI as it doesn't have the required data for the provider table

That is not what I meant. Sorry, should have been clearer. When I created an offer (and forwarded bc), the cache did not capture this event from the BC. I've done this after making sure both cache and ui have the same (and the correct) SC addresses.

**Pull request merged**